model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -1206,6 +1210,15 @@ message MavenPayload {
   string repository_url = 2;
 }
 
+message DeferredArtifactPayload {
+  // A unique string identifier assigned by the creator of this payload. The creator may use this key to confirm
+  // whether they can parse the data.
+  string key = 1;
```

Comment:
Should this be uid? Any collisions here could be bad...

Reply:
This is going to have to get revamped for XLang and since it isn't being exported outside of the SDK for portable runners we can easily change it.